demo/config.js

```diff
@@ -752,7 +762,9 @@ shakaDemo.Config = class {
   addNumberInput_(name, valueName, canBeDecimal = false, canBeZero = true,
       canBeUnset = false, tooltipMessage) {
     const onChange = (input) => {
-      shakaDemoMain.resetConfiguration(valueName);
+      if (valueName !== 'streaming.segmentPrefetchLimit') {
```

Comment:
Why?

Reply:
When testing the prefetch limit in the demo page, this always resets the value before setting it again. However, for the segment prefetch, if you're changing the value but not resetting it completely, this ends up emptying the segment prefetches which makes it harder to test.
That is, if segment prefetch is 5, and we want to go to 10, we still end up clearing out all the segments without this.

However, happy to revert this change or change it in some way, if you have a better idea of how to do this.

Reply:
@theodab Any idea?

Reply:
Initial thought is you could rewrite the function to be something like this:

```js
      if (input.value == 'Infinity') {
        shakaDemoMain.configure(valueName, Infinity);
      } else if (input.value == '' && canBeUnset) {
        shakaDemoMain.resetConfiguration(valueName);
      } else {
        const valueAsNumber = Number(input.value);
        if (valueAsNumber == 0 && !canBeZero) {
          shakaDemoMain.resetConfiguration(valueName);
        } else if (isNaN(valueAsNumber)) {
          shakaDemoMain.resetConfiguration(valueName);
        } else {
          if (Math.floor(valueAsNumber) != valueAsNumber && !canBeDecimal) {
            shakaDemoMain.resetConfiguration(valueName);
          } else {
            shakaDemoMain.configure(valueName, valueAsNumber);
          }
        }
      }
      shakaDemoMain.remakeHash();
```

I haven't tested this extensively, but I think it works. This way, each call to `onChange` will result in exactly one re-configure, rather than un-setting and then re-setting the value.

Reply:
seems to be working as expected. Updated.